include/psp2/registrymgr.h

```diff
@@ -82,6 +82,103 @@ int sceRegMgrSetKeyInt(const char* category, const char* name, int buf);
  */
 int sceRegMgrSetKeyStr(const char* category, const char* name, char* buf, const int size);
 
+/**
+ * Get all key's initial information by category (from os0:kd/registry.db0)
+ *
+ * @param category - The path to the directory to be opened (e.g. /CONFIG/SYSTEM)
+ * @param buf - Pointer to a char buffer to hold the values
+ * @param elements_number - The number of elements of the buffer
+ *
+ * @return 0 on success, < 0 on error
+ */
+int sceRegMgrGetInitVals(const char* category, char* buffer, const int elements_number);
+
+/**
+ * Get all key's information by category
+ *
+ * @param category - The path to the directory to be opened (e.g. /CONFIG/SYSTEM)
+ * @param buf - Pointer to a char buffer to hold the values
+ * @param elements_number - The number of elements of the buffer
+ *
+ * @return 0 on success, < 0 on error
+ */
+int sceRegMgrGetKeys(const char* category, char* buffer, const int elements_number);
+
+/**
+ * Set all key's information by category
+ *
+ * @param category - The path to the directory to be opened (e.g. /CONFIG/SYSTEM)
+ * @param buf - Pointer to a char buffer that holds the values
+ * @param elements_number - The number of elements of the buffer
+ *
+ * @return 0 on success, < 0 on error
+ */
+int sceRegMgrSetKeys(const char* category, char* buffer, const int elements_number);
+
+//
+/**
+ * Get a system param key's information by id
+ *
+ * @param id - The id of the key
+ * @param buf - Pointer to a char buffer to hold the value
+ * @param size - The size of the buffer
+ *
+ * @return 0 on success, < 0 on error
+ */
+int sceRegMgrSystemParamGetInt(const int id, int* buf);
+
+//SceRegMgrForGame_877ADB3F(0x008F94F9, v5 + 0xB0, 3);
+/**
+ * Get a system param key's information by id
+ *
+ * @param id - The id of the key
+ * @param buf - Pointer to a char buffer to hold the value
+ * @param size - The size of the buffer
+ *
+ * @return 0 on success, < 0 on error
+ */
+int sceRegMgrSystemParamGetStr(const int id, char* buf, const int size);
+
+
+/**
+ * Get the registry version
+ *
+ * @param id - The id of the key
+ * @param buf - Pointer to a char buffer to hold the value
+ *
+ * @return 0 on success, < 0 on error
+ */
+int sceRegMgrGetRegVersion(int version, char* buf);
+
+
+/**
+ * Registers a registry callback
+ *
+ * @return 0 on success, < 0 on error
+ */
+int sceRegMgrRegisterCallback();
+
+//v5 = "/CONFIG/DATE/date_format";
+//a1 = SceRegMgr_A86F0A71(0, v5);
+/**
+ * Adds a registry callback
+ *
+ * @param unk - unknown (maybe callback_id)
+ * @param key_path - The path to the key to be callbacked (e.g. "/CONFIG/DATE/date_format")
+ *
+ * @return 0 on success, < 0 on error
+ */
+int sceRegMgrAddRegistryCallback(int unk, const char* key_path);
+
+/**
+ * Adds a registry callback
+ *
+ * @param unk - unknown (same as sceRegMgrAddRegistryCallback arg1)
```

Comment:
`::sceRegMgrAddRegistryCallback`

Reply:
???
I'm not sure if it is the callback id used in sceRegMgrAddRegistryCallback.
but according to reversing it is.
what does mean "::" ?

Reply:
It allows to reference another function so that it can be clicked.
Here is an example: https://github.com/vitasdk/vita-headers/blob/b581e43/include/psp2/ctrl.h#L24